server/ast/expr.go

```diff
@@ -0,0 +1,517 @@
+// Copyright 2023 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ast
+
+import (
+	"fmt"
+	"go/constant"
+
+	vitess "github.com/dolthub/vitess/go/vt/sqlparser"
+
+	"github.com/dolthub/doltgresql/postgres/parser/sem/tree"
+)
+
+// nodeExprs handles tree.Exprs nodes.
+func nodeExprs(node tree.Exprs) (vitess.Exprs, error) {
+	if len(node) == 0 {
+		return nil, nil
+	}
+	exprs := make(vitess.Exprs, len(node))
+	for i := range node {
+		var err error
+		if exprs[i], err = nodeExpr(node[i]); err != nil {
+			return nil, err
+		}
+	}
+	return exprs, nil
+}
+
+// nodeCompositeDatum handles tree.CompositeDatum nodes.
+func nodeCompositeDatum(node tree.CompositeDatum) (vitess.Expr, error) {
```

Comment:
I think some of these are used in the `"not yet supported"` stuff. We could remove them, but if someone later has a `tree.Constant` and doesn't know that they need to pass it to `nodeExpr`, then they'll create a `nodeConstant` and switch on the constant types, which would then create two places with potentially different logic. That's happened a decent amount in GMS, and I want to avoid that here, except in cases where nodes are contextually different.